apps/sms/services/js/messaging/messaging_service.js

```diff
@@ -65,6 +70,20 @@ var MessagingService = {
     this.initService();
   },
 
+  register(appInstanceId) {
+    var mobileMessageClient = MozMobileMessageClient.forApp(appInstanceId);
+
+    EVENTS.forEach((event) => {
+      mobileMessageClient.on(event, (data) => {
+        // Workaround to avoid the duplicated events by inserting appInstanceId
+        // into event name in bridge mixin.
```

Comment:
From main instance and activity instance if it exists. In this case iframe in both instances will fire the event to messaging service, so both instance will receive same message event twice. In this workaround we gave the event with specific app instance id to make sure the event only triggered once to the corresponding instance.